webpack/assets/javascripts/react_app/components/HostStatuses/Status/index.js

```diff
@@ -0,0 +1,203 @@
+import React, { useState, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import {
+  Card,
+  CardHeader,
+  CardExpandableContent,
+  Grid,
+  GridItem,
+} from '@patternfly/react-core';
+import GlobalStatusIcon from './GlobalStatusIcon';
+import { sprintf, translate as __ } from '../../../common/I18n';
+import Details from './Details';
+import LinkOrLabel from './LinkOrLabel';
+import {
+  GLOBAL_STATUS_OK,
+  GLOBAL_STATUS_WARN,
+  GLOBAL_STATUS_ERROR,
+} from './constants';
+
+const Status = ({
+  name,
+  description,
+  ok_total_path: okTotalPath,
+  ok_owned_path: okOwnedPath,
+  warn_total_path: warnTotalPath,
+  warn_owned_path: warnOwnedPath,
+  error_total_path: errorTotalPath,
+  error_owned_path: errorOwnedPath,
+  details,
+}) => {
+  const globalStatus = useMemo(
+    () =>
+      Math.max(
+        ...details
+          .filter(({ total }) => total > 0)
+          .map(({ global_status: gs }) => gs),
+        0
+      ),
+    [details]
+  );
```

Comment:
data manipulation should be done via selectors. 
```js
// StatusSelectors.js
import { selectAPIResponse } from '/API/APISelectors';

export const selectManipulatedStatus = (state, key) =>
   selectAPIResponse(state, key)
    .filter(({ total }) => total > 0)
    .map(({ global_status: gs }) => gs)
}

// index.js
import { selectManipulatedStatus } from './StatusSelectors'
  const {
     status,
     key,
   } = useAPI('get', url});
  const data =  useSelector((state, key) =>  selectManipulatedStatus(state, key)
  // return <Status> with calculated data
```

In some cases, like the useMemo, we can memorize selectors by [reselect](https://github.com/reduxjs/reselect), but we should consider this as optizimation which needs to be done only if we encounter performances issues.

This concrean related also to the below data manipulation.